pkgs/repo_query/lib/labels_update.dart

```diff
@@ -0,0 +1,312 @@
+        }
+      }
+
+      var adds = templateSet.difference(labelsEncountered).map((name) {
+        var templateLabel = templateLabels.firstWhere((l) => l.name == name);
+        return LabelEdit(
+          newName: name,
+          color: templateLabel.color,
+          description: templateLabel.description,
+        );
+      });
+
+      // renames
+      for (var entry in edits.entries.where((e) => e.value.newName != null)) {
+        print('  (rename) ${entry.key}: ${entry.value}');
+      }
+
+      // updates
+      for (var entry in edits.entries.where((e) => e.value.newName == null)) {
+        print('  (update) ${entry.key}: ${entry.value}');
+      }
+
+      // adds
+      for (var edit in adds) {
+        print('  (add) ${edit.newName}: #${edit.color} "${edit.description}"');
+      }
+
+      // advisories
+      for (var entry in advisories.entries) {
+        print('  (consistency) ${entry.key}: ${entry.value}');
+      }
+
+      if (alsoFix) {
+        print('');
+        print('Updating labels for $slug');
+
+        var repoSlug = RepositorySlug.full(slug);
+        var repo =
+            await reportRunner.github.repositories.getRepository(repoSlug);
+        print('  $slug has ${repo.openIssuesCount} issues and '
+            '${labels.length} labels.');
+
+        if (slug == templateRepoSlug || slug == 'dart-lang/sdk') {
+          print("  skipping: won't update labels for $slug.");
+        } else if (repo.openIssuesCount >= 100) {
+          print("  skipping: won't update labels when issue count >=100.");
+        } else {
+          // Perform updates.
+          for (var entry in edits.entries) {
+            var oldName = entry.key;
+            var edit = entry.value;
+
+            print('  updating $oldName: $edit');
+            await reportRunner.github.issues.updateLabel(
+              repoSlug,
+              oldName,
+              newName: edit.newName,
+              color: edit.color,
+              description: edit.description,
+            );
+          }
+
+          for (var edit in adds) {
+            print('  adding ${edit.newName}');
+            await reportRunner.github.issues.createLabel(
+              repoSlug,
+              edit.newName!,
+              color: edit.color,
+              description: edit.description,
+            );
+          }
+
+          print('');
+          print('  ${repo.htmlUrl}/labels');
+        }
+      }
+    }
+
+    return 0;
+  }
+}
+
+String? checkSynonym(String label) {
+  for (var entry in synonyms.entries) {
+    if (entry.value.contains(label.toLowerCase())) {
+      return entry.key;
+    }
+  }
+
+  return null;
+}
+
+String? wellFormed(String label) {
+  if (allowList.contains(label)) return null;
+
+  if (label.startsWith('package:')) {
+    const avoidPrefix = 'package: ';
+    if (label.startsWith(avoidPrefix)) {
+      return 'rename to package:${label.substring(avoidPrefix.length)}';
```

Comment:
Is this to filter out the whitespace after the colon? I would then rather write
```
const packagePrefix = 'package:';
if (label.startsWith(packagePrefix)) {
    var labelEnding = label.substring(avoidPrefix.length);
    if (labelEnding.startsWith(r' \s*')) {
      return 'rename to "package:${labelEnding.trim()}"';
```
But it also seems like whitespaces are caught further down, so this case could also simply be removed?

Reply:
+1, I'll change to using trim() here for the package name

> But it also seems like whitespaces are caught further down, so this case could also simply be removed?

Here, we're trying to find a good value to rename the label to; below, we're just complaining that they're using a space in the label name.